File: duplicatesfromsortedArray.java

// T.C =O(n)
// S.C = O(1)
class Solution {
    public int removeDuplicates(int[] nums) {
        int count = 1;
        int j = 1;
        for (int i=1;i<nums.length;i++)
        {
            
            if(nums[i]==nums[i-1])
            {
                count++;
            }
            else
            {
                count=1;
               
            }
            if(count <= 2)
            {
                nums[j++] = nums[i];
                
            }
        }
        return j;
        
    }
}

File: DuplicatesFromSortedArray.py
# Time Complexity : o(n)
# Space ComplexityL:0(1)
# Did it run on leetcode:Yes
# Any problem you faced while coding this :No
# Solved the problem by maintaining two pointers and a counter


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
#         track insertion position
        i =1 
#     track curr pos in array
        j = 1
        counter = 1
        
        while(j < len(nums)):
            
            if nums[j] == nums[j-1]:
                counter +=1
            else:
                counter = 1
            if (counter <=2):
                nums[i] = nums[j]
                i = i+1
            j = j+1
            
        return i
                
            